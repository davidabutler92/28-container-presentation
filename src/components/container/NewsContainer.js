import React, { useState } from 'react';
import ArticleList from '../ArticleList/ArticleList';
import SeachForm from '../SearchForm/SearchForm';

export default function NewsContainer() {
  const [articles, setArticles] = useState([]);
  const [loading, setLoading] = useState(false);

  return (
    <main>
      <SeachForm setArticles={setArticles} setLoading={setLoading} />
      {loading ? <div>loading</div> : <ArticleList articles={articles} />}
    </main>
  );
}
