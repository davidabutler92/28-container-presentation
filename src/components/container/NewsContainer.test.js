import React from 'react';
import { render, cleanup } from '@testing-library/react';
import NewsContainer from './NewsContainer';

describe('NewsContainer component', () => {
  afterEach(() => cleanup());
  it('renders SearchForm and ArticleList', () => {
    const { asFragment } = render(<NewsContainer />);
    expect(asFragment()).toMatchSnapshot();
  });
});
