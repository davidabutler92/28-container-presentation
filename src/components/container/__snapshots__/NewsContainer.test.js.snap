// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContainer component renders SearchForm and ArticleList 1`] = `
<DocumentFragment>
  <main>
    <form>
      <input
        placeholder="Search"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Search
      </button>
    </form>
    <ul />
  </main>
</DocumentFragment>
`;
